### ISSUE TITLE
The update of Thor and the tools I use hung up today and would not complete

### ISSUE THREAD

**Issue by GTGeek88** [VFPX/Thor#281]
Not sure what else to say. I just chose Check For Updates from the Thor menu and let it rip. Did fine until seemingly right at the very end. Got the hourglass cursor that wouldn't go away and what you see in the screenshot.

**Comment by GTGeek88** [comment]
Had to End Task on VFP, which shutdown immediately. That would seem to indicate VFP was still taking input from Windows, saw the request, and obliged. I restarted VFP, ran the update again, Thor was the only thing that indicated an update was needed and it ran successfully.

**Comment by Jimrnelson** [comment]
@GTGeek88 

Has this problem occurred again?

**Comment by Jimrnelson** [comment]
If the problem with GoToDefinition occurs again, please report it at [https://github.com/VFPX/GoToDefinition/issues](url) with as much information as possible for me to try to replicate it.

**Comment by GTGeek88** [comment]
Jim, the problem with Go To Definition intermittently not coming up to let me add a new property or method happened enough that I feel I'd have seen it by now if the latest version had not fixed it. So, preliminarily, at least, I think the problem is fixed or much reduced. I have not seen the problem at all, so hopefully that bug was completely squashed.

**Comment by Jimrnelson** [comment]
@GTGeek88 

Good news then!  I will mark this issue as closed.  

Should you see any recurrence, please supply as much info as possible so that I can try to reproduce.

**Comment by GTGeek88** [comment]
Can we reopen this? I wondered if I was going to jinx this. I just put this.LastPmtId = this.LastPmtId + 1 in code and double-clicked LastPmtId and chose Go To Definition and nothing happened. Just zilch. Nada. I don't know what else to tell you. Don't know of anything I've done that could affect this.

**Comment by Jimrnelson** [comment]
@GTGeek88 

> Can we reopen this? I wondered if I was going to jinx this. I just put this.LastPmtId = this.LastPmtId + 1 in code and double-clicked LastPmtId and chose Go To Definition and nothing happened. Just zilch. Nada. I don't know what else to tell you. Don't know of anything I've done that could affect this.

Do you have PEMEditor open when you do this?

The only effect you can expect when using GoToDef on a property is that it will _attempt_ to go to that property in PEMEditor if you are in a method in a SCX or VCX.  In a PRG, this should not do anything.

**Comment by GTGeek88** [comment]
No, nothing at all. Definitely no message that looks like your example.

"The only effect you can expect when using GoToDef on a property . . ."

I don't understand. I've added properties many times this way. Anyway, later, after closing the SCX and building and running my program and getting an error that the property did not exist (because I'd stopped to let you know it happened again the forgot to add the property "manually"), I opened the SCX and double-clicked the property name to highlight it and used Go To Definition and the form came right up and I added it. So, go figure. Intermittent problems are always such a joy to track down. And this never happens on just one form or class. It happens on most any form or class. And I still think it has lessened, but I guess it's not fully fixed. I'll keep watching it.

**Comment by Jimrnelson** [comment]
Head scratching time, so here are a few stabs in the dark.  (I'm a little at a loss to know what to ask)

(1) Does the problem with GTD not working only fail (i.e., do nothing) when you write `This.SomeName`?  That is, when it fails, do other uses of GTD still work?

(2) When it fails on `This.SomeName`, have you tried to see whether `ThisForm.SomeName` works?

(3) When it fails, can you use GTD on some other name (maybe some work in a character string) to see if it works ("working", in this sense, might mean displaying a MessageBox -- which disappears after a few seconds).

(4) I am wondering if it's failing because it's not even being called. You can test this by enabling Thor Debug Mode when you are have problems with it:

![image](https://github.com/user-attachments/assets/deb1a363-403e-41c4-9546-03fc89a0c9cf)

If it IS being called, then you should see this when you try GoToDef:

![image](https://github.com/user-attachments/assets/0b2ddcb7-1c15-46df-92b8-8e8c233ade9e)

(You'll want to turn off debug mode once you've tried this -- it's really not of any value otherwise.)

**Comment by GTGeek88** [comment]
Jim,

It's been mostly good. Quite reliable, actually. But I did have two occasions here recently where it failed. The second one was just now. I had typed in this.SavePayment(), double-clicked SavePayment and selected Go to definition from the menu. Nothing. So I looked back over your last communication and saw that I should try thisform. Well, that did make it work. I can't say it's always about that, because I do try to use "thisform" only when the object is not the form and I use "this" when I'm in a method of the form. So if that were the only cause, I think I'd see it a lot more. But this time, using thisform did the trick.

**Comment by Jimrnelson** [comment]
@GTGeek88 

I have created an issue for this on the [GoToDefinition issues page](https://github.com/VFPX/GoToDefinition/issues/35)
